pkg/sessiontxn/isolation/base.go

```diff
@@ -284,6 +284,8 @@ func (p *baseTxnContextProvider) ActivateTxn() (kv.Transaction, error) {
 		return nil, err
 	}
 
+	txn.SetOption(kv.ResourceGroupTagger, p.sctx.GetSessionVars().StmtCtx.GetResourceGroupTagger())
```

Comment:
I'm actually not sure whether this is the proper place to do that. And should `RPCInterceptor` be also set? Or should `setOptionForTopSQL` be used instead?

Reply:
I put it here to let it work as a backup plan when some auto-commit pipelined dml stmts do not have such option set. It can be override later in other places. Or should we put it in `ExecuteStmt()` after `Compile()`?  @crazycs520 @mornyx What do you think?

Reply:
If the committer is already inited here, `SetOption(kv.ResourceGroupTagger, ...)` won't affect the committer.

https://github.com/tikv/client-go/blob/0606e74e8e373d224ad6e1392729b9780b76a8e8/txnkv/transaction/txn.go#L309-L312

Reply:
The pipelined committer is created when txn is created. It seems we need to modify SetOption

Reply:
LGTM

Reply:
> I'm actually not sure whether this is the proper place to do that. And should RPCInterceptor be also set? Or should setOptionForTopSQL be used instead?

I copied the code to set these options before `commit`, and only apply them for pipelined txns, in favor of caution.